apps/fft/fft-stdexec.cpp

```diff
@@ -0,0 +1,221 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2023 The Regents of the University of California,
+ * through Lawrence Berkeley National Laboratory (subject to receipt of any
+ * required approvals from the U.S. Dept. of Energy).All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/*
+ * commons for the fft codes
+ */
+
+#define FFT_STDEXEC
+#include "fft.hpp"
+
+//
+// fft algorithm
+//
+std::vector<data_t> fft(data_t *x, scheduler auto sch, const int N, const int max_threads, bool debug = false)
+{
+    std::vector<data_t> x_rev(N);
+    std::vector<uint32_t> ind(N);
+
+    data_t *x_r = x_rev.data();
+    uint32_t *id = ind.data();
+
+    int shift = 32 - ilog2(N);
+
+    ex::sender auto twiddle = ex::transfer_just(sch, x_r, x, id)
+        | ex::bulk(N, [=](int k, auto x_r, auto x, auto id){
+            id[k] = reverse_bits32(k) >> shift;
+            x_r[k] = x[id[k]];
+        })
+        | ex::then([](auto &&...){});
```

Comment:
`ex::then([](auto &&...){});` this syntax is new to me, I am curious to learn what does it do here?